src/QuadForm/indefiniteLLL.jl

```diff
@@ -0,0 +1,386 @@
+
+  return re
+end
+
+#=
+  _ker_mod_p(M::MatElem{fmpz},p::Int64) -> Int, MatElem{fmpz}
+
+Compute the kernel of the given matrix $M \mod p$. Return $rk, U$, where
+$rk = dim (ker M \mod p)$ and $U$ an invertible matrix over $\mathbb{Z}$.
+The first $rk$ columns of $U$ span the kernel.
+=#
+function _ker_mod_p(M::MatElem{fmpz},p::Int64)
+  rk, k = kernel(change_base_ring(ResidueRing(ZZ,p),M))
+  U = _complete_to_basis(lift(k[:,1:rk]))
+  reverse_cols!(U)
+
+  return rk, U
+end
+
+################################################################################
+#                           Quadratic Equations
+################################################################################
+
+#=
+    _quadratic_form_solve_triv(G::MatElem{fmpz}; base::Bool = false) 
+                                    -> MatElem{fmpz}, MatElem{fmpz}, MatElem{fmpz}
+
+
+Try to compute a non-zero vector in the kernel of $G$ with small coefficients.
+`G` must be a square-matrix with $det(G) = 1$ and dimension at most 6. 
+  Return $G,I,sol$ where $I$ is the identity matrix and $sol$ is non-trivial
+norm 0 vector or the empty vector if no non-trivial vector is found.
+  If base = true and and a norm 0 vector is obtained, return $H^T * G * H$, $H$ and
+$sol$ where $sol$ is the first column of $H$ with norm 0.
+=#
+function _quadratic_form_solve_triv(G::MatElem{fmpz}; base::Bool = false, check::Bool = false)
+  
+  if check == true
+    if det(G) != 1 || ncols(G) != nrows(G) || ncols(G) > 6
+      error("G has to be a unimodular matrix with dimension at most 6.")
+    end
+  end
+  
+  n = ncols(G)
+  H = identity_matrix(base_ring(G),n)
+
+  #Case 1: A basis vector is isotropic
+  for i = 1:n
+    if(G[i,i] == 0)
+      sol = H[:,i]
+      if(base == false)
+        return G, H, sol
+      end
+      H[:,i] = H[:,1]
+      H[:,1] = sol
+
+      return transpose(H)*G*H, H, sol
+    end
+  end
+
+  #Case 2: G has a block +- [1 0 ; 0 -1] on the diagonal
+  for i = 2:n
+    if(G[i-1,i] == 0 && G[i-1,i-1]*G[i,i] == -1)
+  
+      H[i-1,i] = -1
+      sol = H[:,i]
+      if (base == false)
+        return G, H, sol
+      end
+      H[:,i] = H[:,1]
+      H[:,1] = sol
+        return transpose(H)*G*H, H, sol
+    end
+  end
+
+  #Case 3: a principal minor is 0
+  for i = 1:n
+    GG = G[1:i,1:i]
+    if(det(GG) != 0)
+      continue
+    end
+    sol = kernel(GG)[2][:,1]
+    sol = divexact(sol,content(sol))
+    sol = vcat(sol,zero_matrix(base_ring(sol),n-i,1))
+    if (base == false)
+      return G, H, sol
+    end
+    H = _complete_to_basis(sol)
+    H[:,n] = - H[:,1]
+    H[:,1] = sol
+    
+    return transpose(H)*G*H, H, sol
+  end
+
+  return G,H, fmpz[]
+end
+
+###############################################################################
+#                           Quadratic Forms Reduction
+################################################################################
+
+@doc Markdown.doc"""
+    lll_gram_indef(G::MatElem{fmpz}, base::Bool = false) 
```

Comment:
```suggestion
    lll_gram_indef(G::MatElem{fmpz}; base::Bool = false) 
```
We need to describe what the `base` keyword does.